### PR TITLE
Adding placeholder support for {{i18n}} helper and i18n support for server-side JS

### LINKED ISSUE #166: Placeholder support for 'i18n' helper
Currently`{{i18n}}` helper doesn't support resource string with placeholders (variables).

Example resource entry: `message="User %s cannot be found."`

Should be able to pass the values for the placeholders when we use `{{i18n}}` helper.

### PR DESCRIPTION
**Pull request by cnapagoda**
adding placeholder support for 'i18n' helper and server-side i18n support.

RESOLVE #166 #160

### CODE REVIEW

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/I18nResources.java [hunk 1]
```diff
@@ -62,10 +66,42 @@ public void addI18nResource(String language, Properties i18n) {
...
-     * @param language language of the i18n resource
+     * @param locale Locale of the i18n resource
      * @return i18n resource
      */
-    public Properties getI18nResource(String language) {
-        return i18nResources.get(language);
+    public Properties getI18nResource(Locale locale) {
+        return i18nResources.get(locale.toLanguageTag());
+    }
+
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/I18nResources.java]: Remove unnecessary new line.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/AppCreator.java [hunk 2]
```diff
@@ -270,7 +271,14 @@ private void addI18nResources(Map<String, Properties> componentI18nResources, I1
         if ((componentI18nResources == null) || componentI18nResources.isEmpty()) {
             return;
         }
-        componentI18nResources.forEach(i18nResources::addI18nResource);
+        componentI18nResources.forEach((localString, properties) -> {
+            Locale locale = Locale.forLanguageTag(localString.replace("_", "-"));
+            if (!locale.getLanguage().isEmpty()) {
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/AppCreator.java]: Let's invert this if condition, in order to eliminate the not operator.

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/AppCreator.java [hunk 3]
```diff
@@ -270,7 +271,14 @@ private void addI18nResources(Map<String, Properties> componentI18nResources, I1
...
+                i18nResources.addI18nResource(locale, properties);
+            } else {
+                throw new UUFException("Cannot deploy language file for " + localString);
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/internal/deployment/AppCreator.java]: Shall we add a more detailed error message here. Clearly we NOT deploying language files.

File: components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/core/js/I18nFunction.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (this) [inline components/uuf-renderablecreator-hbs/src/main/java/org/wso2/carbon/uuf/renderablecreator/hbs/core/js/I18nFunction.java]: Year 2017

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/I18nResources.java (hunk 1 above)
Comment (cnapagoda) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/I18nResources.java]: fixed with 	b9efe27

**Pull request merged**